src/geobr.jl

```diff
@@ -0,0 +1,251 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+module GeoBR
+
+using GeoIO
+using DataDeps
+using CSV
+using Tables
+using TableTransforms
+
+const APIVERSIONS = (v"1.7.0",)
+
+function extract_metadata(geo, year, code, code_abbrev; return_all=false)
+    table = CSV.File(joinpath(@__DIR__, "..", "artifacts", "GeoBR.csv"))
+
+    srows = table |> Filter(row ->
+        row.geo == geo &&
+            row.year == year &&
+            (isnothing(code) || parse(Int, row.code) == code) &&
+            (isnothing(code_abbrev) || row.code_abbrev == code_abbrev)
+    )
+
+    if isempty(srows)
+        throw(ErrorException("No matching rows found for the given parameters"))
+    end
+
+    return return_all ? Tables.rows(srows) : Tables.rows(srows) |> first
+end
+
+function download(geo, year, code, code_abbrev)
+    srow = extract_metadata(geo, year, code, code_abbrev)
+
+    url = srow.download_path
+    fname = split(url, "/") |> last |> splitext |> first
+    ID = "GeoBR_$fname"
+    try
+        # if data is already on disk
+        # we just return the path
+        @datadep_str getdatadep(ID, url)
+    catch
+        # otherwise we register the data
+        # and download using DataDeps.jl
+        try
+            register(DataDep(ID,
+                """
+                Metadata for GeoBR package.
+                Source: $url
+                """,
+                url,
+                Any
+            ))
+            @datadep_str getdatadep(ID, url)
+        catch
+            throw(ErrorException("download failed due to internet and/or server issues"))
+        end
+    end
+end
+
+function get(geo, year, code=nothing, code_abbrev=nothing; kwargs...)
+    path = download(geo, year, code, code_abbrev)
+    GeoIO.load(path; kwargs...)
+end
+
+function state(state; year=2010, kwargs...)
+    code = isa(state, Number) ? state : nothing
+    code_abbrev = isa(state, AbstractString) ? state : nothing
+    gdf = get("state", year, code, code_abbrev, kwargs...)
+    return gdf
```

Comment:
Julia returns the last statement by default
```suggestion
    get("state", year, code, code_abbrev, kwargs...)
```